plugin/pulpcore/plugin/download/base.py

```diff
@@ -83,6 +86,10 @@ def __init__(self, url, custom_file_object=None, expected_digests=None, expected
             self.path = self._writer.name
         self.expected_digests = expected_digests
         self.expected_size = expected_size
+        if semaphore:
+            self.semaphore = semaphore
+        else:
+            self.semaphore = asyncio.Semaphore()  # This will always be acquired
```

Comment:
This could be simplified to ```self.semaphore = semaphore or asyncio.Semaphore()``` but this way might be more clear to readers, up to you.

Reply:
I was trying to remember that idiom! Thanks @dralley

Reply:
I like your simplified idiom personally :)

Reply:
Actually I'm going to keep it as is because even though I like the shortened idiom, this for lets me keep the comment which I think is helpful since creating a random Semaphore might not be obvious why.